roles/build_project_repo/tasks/python_app_version.yml

```diff
@@ -0,0 +1,41 @@
+# Determine the version of the python app being deployed
+- name: Determine python app version
+  tags:
+    - python
+  block:
+    - name: Check if project has a version.py file
+      stat:
+        path: "{{ clone_root}}/{{ repo }}/{{ python_app }}/version.py"
```

Comment:
Pretty sure that the version.py file should be inside the package and not floating freely at the top level of the repo.  I wondered about the paths when you revised this.

Importing just the version without importing the main package shouldn't trigger the other imports - you ought to be able to check that fairly easily. Could you confirm and then move the version file and update the other references?

fwiw, IMO the `import *` you have in geotaste.__init__ is a bad pattern - I wasn't paying attention enough last week to understand that's what we're working around here

Reply:
I see Henry recommended using `_version.py` - is that because it's meant as an internal that others shouldn't be importing from directly? 

I'm ok with switching to that if you prefer it. But either way it's still my understanding that it needs to be inside the package.

Reply:
I just think for a general purpose build_project_repo we should make as few assumptions about the importing logic of the package we're building. I'm ok with putting it in package folder but I would like to try then using importlib to import that file directly by path.

Reply:
I tested and see that you're right, nesting the version.py within the app does trigger the other imports that aren't installed at this point in the deploy.

I feel like we've both spent too much time on this already, and I'm not sure how you're using versioning in this app anyway. The `python_app_path` is a convenience to make the deploy paths more semantic, and I'm not actually sure how you're versioning or plan to version the geotaste app further. 

Let's go with the simplest path that gets you to a working deploy and doesn't break other existing deploys.

Here are a couple of options, but feel free to propose something else:
- refactor new python_app_version task so it matches what you have in your branch 
- skip the logic for getting the python app version from the code:
  - update deploy / path / app version variables so you just set version explicitly and/or don't include version in the deploy path (I think you asked for this initially and I didn't understand why this was a problem for this app, I apologize)
  - add a variable to control whether python app version should be pulled from code; set it to be on by default, off for geotaste